static/app/views/alerts/rules/metric/details/metricHistory.tsx

```diff
@@ -155,14 +64,24 @@ function MetricHistory({incidents}: Props) {
             emptyMessage={t('No alerts triggered during this time.')}
             expanded={numOfIncidents <= COLLAPSE_COUNT || isExpanded}
           >
-            {filteredIncidents.map((incident, idx) => {
+            {sortedActivity.map((item, idx) => {
               if (idx >= COLLAPSE_COUNT && !isExpanded) {
                 return null;
               }
+              if ('activator' in item) {
+                return (
+                  <MetricAlertActivation
+                    key={`${item.type}-${item.activator}`}
+                    activationActivity={item}
+                    organization={organization}
+                  />
+                );
+              }
+
               return (
                 <MetricAlertActivity
                   key={idx}
-                  incident={incident}
+                  incident={item}
                   organization={organization}
                 />
               );
```

Comment:
I'd pull this out as a separate component, `MetricAlertActivityList` that takes a list of `activities` and renders them using a switch statement:
```js
function MetricAlertActivityList({ activities }) {
    return activities.map((activity, idx) => {
       let Activity;
       // rather than looking to see if it has an activator, it'd be cool if we could check the alert type instead to see if it's an activated alert or a standard activation.
       switch(activity.type) {
           case AlertRuleType.ACTIVATED: 
             Activity = MetricAlertActivationActivity;
          default:
             Activity = MetricAlertActivity;
       }
       return <Activity key={key} incident={activity} organization={organization} />
    });
}
```
This would need to change the API for `MetricAlertActivation` and rename it to `MetricAlertActivationActivity`. The nice thing is that if we have new alert types in the future or want to revisit we can simply add a new type of activity and render that one here. Extra credit could be to create a base interface for activities to share.

Reply:
I do think there _is_ such a thing as too micro of components here :/ 
and in this particular case, that's essentially waht this component is... if i were to pull out the actual map into its own component, then the `<metricHistory>` component could just be squashed into the <body /> component 🤷 
so... going to defer this suggestion

I agree though, i'm cheating a little bit in that i'm only really sneaking in a new "type" of rendered component here (the activation history). 
the "incident history" that we're rendering here _does not_ have a "type" perse... I'm not quite sure how to explain it, but Incidents and Activations are just different "row type"s - NOT synonymous with any data model 'alert type' or anything like that.
Simply just a row in the table.

Reply:
> I do think there is such a thing as too micro of components here :/

I actually don't. Think like atomic design; atoms, molecules, organisms. To me, we don't have much in the way of atoms / molecules, which ends up meaning we have to rebuild a lot of logic that is just a little bit different. By decomposing into smaller and smaller pieces we can build up larger molecules, so we can quickly build out organisms as needed from those smaller building blocks. The smaller you can refactor the logic, the more composable and reusable that piece of logic is.

This is a good example of why we'd have to build a new one-off or make unexpected changes in the future. If i'm going to add a new alert type, i'd be surprised if i had to refactor these areas because it's not utilizing the type to differentiate between them.

Reply:
mmm i guess in a more theoretical sense- i might rephrase in that _without proper structure/framework_ I do think there's such thing as too micro of components.
As we have it here today, breaking things up into atoms will just become too messy and end up being sand between our fingers.

This is a theoretical discussion that we don't have to have here though :/ 

For this particular change - again its difficult to explain, but there are no _type_'s here.
This isn't a new alert type at all - and hence there's no 'type' to differentiate between. What we want to differentiate between is simply whether to render an informational row - or simply an 'event' row. (this is completely unrelated to alerts, or the content _in_ these rows)